src/StakeRegistry.sol

```diff
@@ -531,4 +531,67 @@ contract StakeRegistry is VoteWeigherBase, StakeRegistryStorage {
         return _totalStakeHistory[quorumNumber].length;
     }
+
+    /**
+     * @notice Returns a list of all strategies that are restakeable for the middleware
+     * @dev This function is used by an offchain actor to determine which strategies are restakeable
+     */
+    function getRestakeableStrategies() external view returns (IStrategy[] memory){
+        if(quorumBitmap == 0) {
+            return new IStrategy[](0);
+        }
+
+        bytes memory quorumNumbers = BitmapUtils.bitmapToBytesArray(quorumBitmap);
+        IStrategy[] memory strategies = new IStrategy[](_getNumStrategiesInBitmap(quorumNumbers));
+        uint256 index = 0;
+        for(uint256 i = 0; i < quorumNumbers.length; i++){
+            StrategyAndWeightingMultiplier[] memory strategiesAndMultipliers = strategiesConsideredAndMultipliers[uint8(quorumNumbers[i])];
+            for(uint256 j = 0; j < strategiesAndMultipliers.length; j++){
+                strategies[index] = strategiesAndMultipliers[j].strategy;
+                index++;
+            }
+        }
+
+        return strategies;
+    }
+
+    /**
+     * @notice Return a list of all strategies and corresponding share amounts for which the operator has restaked
+     * @dev There may strategies for which the operator has restaked on the quorum but has no shares. In this case,
+     *      the strategy is included in the returned array but the share amount is 0
+     */
+    function getOperatorRestakedStrategies(address operator) external view returns (IStrategy[] memory, uint96[] memory){
+        bytes32 operatorId = registryCoordinator.getOperatorId(operator);
+        uint192 operatorBitmap = registryCoordinator.getCurrentQuorumBitmapByOperatorId(operatorId);
+
+        // Return empty arrays if operator does not have any shares at stake OR no strategies can be restaked
+        if (operatorBitmap == 0 || quorumBitmap == 0) {
+            return (new IStrategy[](0), new uint96[](0));
+        }
+
+        uint192 operatorRestakedQuorumsBitmap = operatorBitmap & quorumBitmap; // get all strategies that are considered restaked
+        bytes memory restakedQuorums = BitmapUtils.bitmapToBytesArray(operatorRestakedQuorumsBitmap);
+        IStrategy[] memory strategies = new IStrategy[](_getNumStrategiesInBitmap(restakedQuorums));
+        uint96[] memory shares = new uint96[](strategies.length);
+
+        uint256 index = 0;
+        for(uint256 i = 0; i < restakedQuorums.length; i++) {
+            StrategyAndWeightingMultiplier[] memory strategiesAndMultipliers = strategiesConsideredAndMultipliers[uint8(restakedQuorums[i])];
+            for (uint256 j = 0; j < strategiesAndMultipliers.length; j++) {
+                strategies[index] = strategiesAndMultipliers[j].strategy;
+                shares[index] = getCurrentOperatorStakeForQuorum(operatorId, uint8(restakedQuorums[i]));
+                index++;
+            }
+        }
+
+        return (strategies, shares);
```

Comment:
i don't think it makes sense to return shares? let's just assume all stake of the strategy is staked until slashing is live